File: modules/IntroWMR/source/original_code/wmr-inst2/wmr-languages/java/mareducer1.java
class Reducer
{
	public void reduce(String key, WmrIterator iter)
	{
		int total = 0;
		int count = 0;
		
		for (String value : iter)
		{
			total += Integer.parseInt(value);
			count++;
		}
		
		double average = (double)total / (double)count;
		Wmr.emit(key, Double.valueOf(average).toString());
	}
}

File: modules/IntroWMR/source/original_code/wmr-inst2/wmr-languages/cpp/mareducer1.cpp
#include <string>
#include <sstream>
#include <iostream>

class reducer
{
public:
	void reduce(std::string key, wmr::datastream stm)
	{
		using std::string;
		using std::stringstream;
		
		unsigned int total = 0;
		unsigned int count = 0;
		unsigned int temp = 0;
		string buffer;
		
		while (!stm.eof())
		{
			stm >> buffer;
			temp = wmr::utility::fromString<unsigned int>(buffer);
			buffer.clear();
			
			total += temp;
			++count;
		}
		
		double average = (double)total / (double)count;
		wmr::emit(key, average);
	}
};
